pkg/utils/utils.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/submariner-io/armada/pkg/defaults"
+)
+
+// ClusterNamesFromFiles will return all clusters from the existing kind files.
+// An error is returned if there's a failure to read the config directory.
+func ClusterNamesFromFiles() ([]string, error) {
+	var clusters []string
+	files, err := ioutil.ReadDir(defaults.KindConfigDir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
+		clusterName := strings.FieldsFunc(file.Name(), func(r rune) bool { return strings.ContainsRune(" -.", r) })[2]
+		clusters = append(clusters, clusterName)
+	}
+
+	return clusters, nil
+}
+
+// ClusterNamesOrAll will return the cluster names sent to it, if there are any.
+// In case the slice is empty, it will read the cluster names from the existing kind files.
+// Should the read fail, it will fatally log the error (causing the process to abort).
+func ClusterNamesOrAll(clusters []string) []string {
+	if len(clusters) > 0 {
+		return clusters
+	}
+
+	clusters, err := ClusterNamesFromFiles()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
I think this should propagate the error for consistency. There's a mix of propagation and use of `log.Fatal` - we should clean that up - I prefer error propagation to the topmost level.

Reply:
I opted to keep the current behavior since it's a refactor to reuse code and not to change the behavior.
If you want to remove log.Fatal usage then we can do it in a separate patch that encompasses the entire project.
Personally I don't like the boilerplate-ness that go error handling and propagation forces upon us, hence I'm not as much against the log.Fatal usage because the alternative is a bunch of "if err -> return err" statements all over the place, and this being a CLI it's not that awful to abort it with exit(1).

Reply:
That's fine to look into that in general for a later patch.

I understand what you're saying about boilerplate but unfortunately that's the way the language designers  decide to handle errors, rather than exceptions.  So I don't think one should necessarily forego  propagation just b/c they don't like how Go implements it. I find it cumbersome too at times but it is what it is.

Either way I think we should be consistent - there's mixed usage of propagation and ` log.Fatal` usage even within he same functions. 

I think embedding `log.Fatal` usage deeper in the code can be detrimental, especially if the function also returns error. There may be cases where a caller can handle an error and proceed w/o exiting  but, by prematurely assuming it's fatal, it doesn't get that chance to decide. So I usually prefer to propagate errors to let the upper layers decide what to do. Also propagating allows intermediate layers to do clean up if they need to.

Another alternative is to use `panic` - this also allows callers to recover if need be but I think it's cleaner to return errors.

Reply:
OK let's discuss this as a more general topic and address this in a PR for the whole code base?